metrics/registry/collector.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package registry
+
+import (
+	"time"
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/common/constant"
+	"dubbo.apache.org/dubbo-go/v3/metrics"
+)
+
+var (
+	rc           registryCollector
+	registryChan = make(chan metrics.MetricsEvent, 128)
+)
+
+func Collector(m metrics.MetricRegistry, c *metrics.ReporterConfig) {
+	rc.regRegistry = m
+}
+
+func init() {
+	metrics.AddCollector("registry_info", Collector)
+	metrics.Subscribe(constant.MetricsRegistry, registryChan)
+	go rc.start()
```

Comment:
I think this init too early ,if metric is not enable, no need to init subscribe and goroutine, maybe move it to collectFunc